File: frontend/src/app/services/hero.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import baseUrl from './helper';
@Injectable({
  providedIn: 'root'
})
export class HeroService {

  constructor(private http : HttpClient ) { }

  public addHero(  hero : any)
  {
    return this.http.post(`${baseUrl}/hero/`,hero);
  }
  public loginHero (login : any)
  {
    return this.http.post(`${baseUrl}/hero/login`,login);
  }
  public findHero (needy: any)

  
  {
    console.log(needy.pincode)
    return this.http.put(`${baseUrl}/hero/${needy.pincode}`,needy);
  }
  public acceptNeedy(heroid :any,needyid:any){
    return this.http.put(`${baseUrl}/hero/accept/${needyid}`,heroid);
  }
  public taskCompleted( needyid :any , heroid:any )
  {
    return this.http.put(`${baseUrl}/hero/completed/${needyid}`,heroid);

  }
  public feedback ( needyid :any, like : any)
  {
    return this.http.put(`${baseUrl}/hero/feedback/${needyid}`,like);
  }
  public catchUpNeedy ( usrName : any)
  {
    return this.http.get(`${baseUrl}/hero/catchupneedy/${usrName}`);
  }
 
}

File: Hero/src/main/java/com/hero/hero/services/HeroService.java
package com.hero.hero.services;

import com.hero.hero.models.Hero;
import com.hero.hero.models.Needy;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.util.ArrayList;
import java.util.HashSet;

@Service
public interface HeroService {
    Hero createHero( Hero hero ) throws Exception;
    Hero loginHero(String usrName ,String password) throws Exception;
    ArrayList<Long> findHero(Long pinCode, Needy needy) throws Exception;
    Hero acceptNeedy(Long needyid,Long heroid ) throws IOException;
//    Hero prepareResponse (Hero hero) ;
    Boolean taskCompleted(Long needyid,Long heroid);
    void feedback(Long needyid ,String like);
    HashSet<Needy> catchUpNeedy(String usrName);
    Boolean editHeroProfile (Hero hero) throws IOException;


}
